### ISSUE TITLE
"Out of sync" error for System.Speech tests on mono in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#46825]
```
C:\h\w\A86A08F9\w\A7BF08CB\e>"C:\h\w\A86A08F9\p\dotnet.exe" exec --runtimeconfig System.Speech.Tests.runtimeconfig.json --depsfile System.Speech.Tests.deps.json xunit.console.dll System.Speech.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing  
  Discovering: System.Speech.Tests (method display = ClassAndMethod, method display options = None)
cant resolve internal call to "System.Speech.Internal.SapiInterop.SpDataKey::.ctor" (tested without signature also)

Your mono runtime and class libraries are out of sync.
The out of sync library is: C:\h\w\A86A08F9\w\A7BF08CB\e\System.Speech.dll
```
e.g.
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-46819-merge-3e8fc30918c1436b9c/System.Speech.Tests/console.76c11600.log?sv=2019-07-07&se=2021-01-31T18%3A38%3A47Z&sr=c&sp=rl&sig=b%2BOWajnPw1h0EEJjMLqteeo5IsjunzIwulVDyEu4iUU%3D
and
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-46819-merge-8cd2e0a3ab684473a3/System.Speech.Tests/console.582451ef.log?sv=2019-07-07&se=2021-01-31T18%3A38%3A47Z&sr=c&sp=rl&sig=dwSi%2FEvmTaVcG%2BDifVZ5yKizLh4QNOyW0vmLIESKoT8%3D

https://dev.azure.com/dnceng/public/_build/results?buildId=945107&view=ms.vss-test-web.build-test-results-tab

cc: @danmosemsft

**Comment by danmosemsft** [comment]
Uh, @steveisok any idea what this might mean? System.Speech.dll is no different to a customer library -- it should run just the same on Mono as on CLR. It uses COM interop but so do other libraries eg System.Management.

**Comment by danmosemsft** [comment]
I would have assumed this just meant the corelib didn't match the Mono runtime, but that wouldn't explain why it calls out Speech explicitly.

**Comment by jkotas** [comment]
>  It uses COM interop but so do other libraries eg System.Management.

Mono does not have full COM interop support. Tests that depend on full COM interop support are disabled for Mono. For example, System.Management tests are disabled against https://github.com/dotnet/runtime/issues/34689

**Comment by steveisok** [comment]
We re-enabled our mono windows build and missed this.  

PR is up https://github.com/dotnet/runtime/pull/46828

**Comment by lateralusX** [comment]
https://github.com/dotnet/runtime/pull/46573 re-enabled Windows build, all was green on that PR but it raced with another PR adding System.Speech in parallel so issue didn't manifest until merged. Mono doesn't support full COM interop on netcore, so we need to disable the System.Speech.Tests on Mono runtime.

**Comment by akoeplinger** [comment]
This was fixed by https://github.com/dotnet/runtime/pull/46828
